### PR TITLE
[drape] Minor constexpr warning fix

### PR DESCRIPTION
**Pull request by biodranik**
... plus some clang-tidying )

### CODE REVIEW

File: drape/support_manager.cpp [hunk 1]
```diff
@@ -139,48 +142,32 @@ bool SupportManager::IsVulkanForbidden(std::string const & deviceName,
       Configuration{"Mali-G72", {1, 1, 131}, {26, 0, 0}},
   };
 
-  for (auto const & d : kBannedDevices)
+  return std::any_of(kBannedConfigurations, kBannedConfigurations + ARRAY_SIZE(kBannedConfigurations), [&](auto const & c)
```
Comment (vng) [inline drape/support_manager.cpp:145]: IMO, before was readable, now is an unreadable long string without any benefits.

File: drape/support_manager.cpp [hunk 2]
```diff
@@ -101,27 +99,32 @@ void SupportManager::ForbidVulkan()
...
     forbidden = false;
   return forbidden;
 }
 
-bool SupportManager::IsVulkanForbidden(std::string const & deviceName,
-                                       Version apiVersion, Version driverVersion) const
+bool SupportManager::IsVulkanForbidden(std::string const & deviceName, Version apiVersion, Version driverVersion)
 {
-  static char constexpr * kBannedDevices[] = {
+  static char constexpr const * kBannedDevices[] = {
```
Comment (vng) [inline drape/support_manager.cpp:112]: Well, constexpr adds only buzzword here.
Probably, you wanted to write something like ```char const * constexpr kBannedDevices``` ... But anyway

File: drape/support_manager.cpp (hunk 1 above)
Comment (biodranik) [reply drape/support_manager.cpp]: Returned back. We have different taste.

File: drape/support_manager.cpp [hunk 3]
```diff
@@ -16,7 +14,7 @@ namespace dp
 {
 struct SupportManager::Configuration
 {
-  std::string m_deviceName;
+  char const * const m_deviceName;
```
Comment (vng) [inline drape/support_manager.cpp:17]: string_view is better here
One day will have a risk to get ```char const * == char const *```

**Review by vng — APPROVED** [review]

**Comment by biodranik** [comment]
Thanks!

**Pull request merged**